src/validation.h

```diff
@@ -851,6 +854,11 @@ class ChainstateManager
     friend CChainState;
 
 public:
+    explicit ChainstateManager(const CChainParams& chainparams) : m_chainparams{chainparams} { }
+
+    const CChainParams& GetParams() const { return m_chainparams; }
+    const Consensus::Params& GetConsensus() const { return m_chainparams.GetConsensus(); }
```

Comment:
4cbc4e94b764ba2b4a79610fadbb092a7bdd076e: What about a public `m_params` and `m_consensus`? There shouldn't be any risk associated with read-only references. And a public member removes the need for a function wrapper that only returns the member.

Reply:
6 to one, half a dozen the other? I thought accessors were more common than extra variables, and make it a bit clearer that they can't get out of sync? *shrug*